tests/unit/auth/permissionChecks.js

```diff
@@ -0,0 +1,268 @@
+const assert = require('assert');
+const { checkBucketAcls, checkObjectAcls } = require('../../../lib/api/apiUtils/authorization/permissionChecks');
+const constants = require('../../../constants');
+
+const { bucketOwnerActions } = constants;
+
+describe('checkBucketAcls', () => {
+    const mockBucket = {
+        getOwner: () => 'ownerId',
+        getAcl: () => ({
+            Canned: '',
+            FULL_CONTROL: [],
+            READ: [],
+            READ_ACP: [],
+            WRITE: [],
+            WRITE_ACP: [],
+        }),
+    };
+
+    const testScenarios = [
```

Comment:
We can also test mixed ACLs like so
```
{
    description: 'should return false if FULL_CONTROL is set for one ID, but WRITE is explicitly denied for the same ID',
    input: {
        bucketAcl: {
            FULL_CONTROL: ['anyId'],
            WRITE: ['anyId'],
        },
        requestType: 'bucketPut',
        canonicalID: 'anyId',
        mainApiCall: 'anyApiCall',
    },
    expected: false,
}
```

Same for log delivery write ACL (logic was unchaged, but let's take advantage of this PR to complete our test suite)
```
{
    description: 'should return true for log-delivery-write ACL when canonicalID matches logId',
    input: {
        bucketAcl: { Canned: 'log-delivery-write' },
        requestType: 'bucketGetACL',
        canonicalID: 'logId',
        mainApiCall: 'anyApiCall',
    },
    expected: true,
}
```

And then the negative case, when no canonical id match

```
{
    description: 'should return false when the canonicalID is not the owner and has no ACL permissions',
    input: {
        bucketAcl: {
            FULL_CONTROL: ['someOtherId'],
            WRITE: ['someOtherId'],
        },
        requestType: 'bucketPut',
        canonicalID: 'anyId',
        mainApiCall: 'anyApiCall',
    },
    expected: false,
}
```

And then one to test a deny if we do a bucketPutACL but we don't have the WRITE_ACP permission:
```
{
    description: 'should return false for bucketPut if canonicalID does not have WRITE_ACP permission',
    input: {
        bucketAcl: {
            WRITE_ACP: ['someOtherId'],
        },
        requestType: 'bucketPutACL',
        canonicalID: 'anyId',
        mainApiCall: 'anyApiCall',
    },
    expected: false,
}
```

(this is not tested)

Reply:
Tests were added here : https://github.com/scality/cloudserver/pull/5420/commits/6cb226cbd6669c89fca2a622d42f9e645c453a2c

- For the bucketPut as it will always return a `false` => the test added is with ACL permissions 
- Added a test for objectPut for when the canonicalID is not the owner and has no ACL permissions 
- Added a test for log-delivery-write ACL when canonicalID matches logId on bucketGetACL
- Added a test for bucketPutACL when canonicalID does not have ACL permissions